autode/opt/optimisers/base.py

```diff
@@ -0,0 +1,608 @@
+                                       species: 'autode.species.Species',
+                                       method:  'autode.wrappers.base.Method'
+                                       ) -> None:
+        """
+        Initialise the internal species and method. They be the correct types
+
+        -----------------------------------------------------------------------
+         Raises:
+             (ValueError): For incorrect types
+         """
+        from autode.species.species import Species
+        from autode.wrappers.base import Method
+
+        if not isinstance(species, Species):
+            raise ValueError(f'{species} must be a autoode.Species instance '
+                             f'but had {type(species)}')
+
+        if not isinstance(method, Method):
+            raise ValueError(f'{method} must be a autoode.wrappers.base.Method '
+                             f'instance but had {type(method)}')
+
+        self._method, self._species = method, species
+        return None
+
+    def _update_gradient_and_energy(self) -> None:
+        """
+        Update the gradient of the energy with respect to the coordinates
+        using the method. Will transform from the current coordinates type
+        to Cartesian coordinates to perform the calculation, then back.
+
+        -----------------------------------------------------------------------
+        Raises:
+            (autode.exceptions.CalculationException):
+        """
+        # Calculations need to be performed in cartesian coordinates
+        self._species.coordinates = self._coords.to('cart')
+
+        grad = Calculation(name=f'{self._species.name}_opt_{self.iteration}',
+                           molecule=self._species,
+                           method=self._method,
+                           keywords=self._method.keywords.grad,
+                           n_cores=self._n_cores)
+        grad.run()
+
+        # Set the energy, gradient and remove all the calculation files
+        self._coords.e = self._species.energy = grad.get_energy()
+        self._species.gradient = grad.get_gradients()
+        grad.clean_up(force=True, everything=True)
+
+        fixed_idxs = self._species.constraints.cartesian
+        self._coords.update_g_from_cart_g(arr=self._species.gradient,
+                                          fixed_atom_idxs=fixed_idxs)
+        return None
+
+    def _update_hessian_gradient_and_energy(self) -> None:
+        """
+        Update the energy, gradient and Hessian using the method. Will
+        transform from the current coordinates type to Cartesian coordinates
+        to perform the calculation, then back. Uses a numerical Hessian if
+        analytic Hessians are not implemented for this method
+
+        -----------------------------------------------------------------------
+        Raises:
+            (autode.exceptions.CalculationException):
+        """
+        self._update_gradient_and_energy()
```

Comment:
This is slightly wasteful if the Hessian calculation also evaluates the gradient, however, gradient evaluations are often a small fraction of the cost of Hessians so _shouldn't_ be too bad